x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/render_cell_value.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import React, { type ReactNode } from 'react';
+import { isDefined } from '@kbn/ml-is-defined';
+import { ALERT_DURATION, ALERT_END, ALERT_START } from '@kbn/rule-data-utils';
+import type { GetRenderCellValue } from '@kbn/triggers-actions-ui-plugin/public';
+import { FIELD_FORMAT_IDS, FieldFormatsRegistry } from '@kbn/field-formats-plugin/common';
+import { getSeverityColor } from '@kbn/ml-anomaly-utils';
+import { EuiHealth } from '@elastic/eui';
+import {
+  alertFieldNameMap,
+  ALERT_ANOMALY_SCORE,
+  ALERT_ANOMALY_TIMESTAMP,
+} from '../../../common/constants/alerts';
+import { getFieldFormatterProvider } from '../../application/contexts/kibana/use_field_formatter';
+
+interface Props {
+  columnId: string;
+  data: any;
+}
+
+export const getMappedNonEcsValue = ({
+  data,
+  fieldName,
+}: {
+  data: any[];
+  fieldName: string;
+}): string[] | undefined => {
+  const item = data.find((d) => d.field === fieldName);
+  if (item != null && item.value != null) {
+    return item.value;
+  }
+  return undefined;
+};
+
+const getRenderValue = (mappedNonEcsValue: any) => {
+  // can be updated when working on https://github.com/elastic/kibana/issues/140819
+  const value = Array.isArray(mappedNonEcsValue) ? mappedNonEcsValue.join() : mappedNonEcsValue;
+
+  if (!isEmpty(value)) {
+    if (typeof value === 'object') {
+      return JSON.stringify(value);
+    }
+    return value;
+  }
+
+  return '—';
+};
+
+export const getRenderCellValue = (fieldFormats: FieldFormatsRegistry): GetRenderCellValue => {
+  const alertValueFormatter = getAlertFormatters(fieldFormats);
+
+  return ({ setFlyoutAlert }) =>
+    (props): ReactNode => {
+      const { columnId, data } = props as Props;
+      if (!isDefined(data)) return;
+
+      const mappedNonEcsValue = getMappedNonEcsValue({
+        data,
+        fieldName: columnId,
+      });
+      const value = getRenderValue(mappedNonEcsValue);
+
+      return alertValueFormatter(columnId, value);
+    };
+};
+
+export function getAlertFormatters(fieldFormats: FieldFormatsRegistry) {
+  const getFormatter = getFieldFormatterProvider(fieldFormats);
+
+  return (columnId: string, value: any): React.ReactElement => {
+    switch (columnId) {
+      case ALERT_START:
+      case ALERT_END:
+      case ALERT_ANOMALY_TIMESTAMP:
+        return <>{getFormatter(FIELD_FORMAT_IDS.DATE)(value)}</>;
+      case ALERT_DURATION:
+        return (
+          <>
+            {getFormatter(FIELD_FORMAT_IDS.DURATION, {
```

Comment:
I use the `fieldFormats` plugin, it supports the following [output formats](https://github.com/elastic/kibana/blob/ad493e1a25101c744a968fac6041ae6ce6d712dc/src/plugins/field_formats/common/constants/duration_formats.ts#L92).

Currently applied `humanizePrecise` works the best. I also tried the `humanize` option, but it ceils values very aggressively, e.g. instead of `1.65 minutes` it becomes `2 minutes`.

Reply:
@mdefazio any thoughts on this one? Wonder if e.g. `2 minutes` is preferable to `1.65 minutes`?

Reply:
Yes, agree if we can make this more readable that woudl be ideal.  Is this rounding also available via an Adv settings option?

Reply:
Let's go with `humanizePrecise` for now - we also use that in the trained models list in some places. But @darnautov can you create an enhancement request for the `field_formats` plugin to improve readability so that there is an option to render e.g. `1.50 minutes` as e.g `1 min 30 sec`